gandi/cli/core/utils/__init__.py

```diff
@@ -156,20 +157,34 @@ def output_snapshot_profile(gandi, profile, output_keys, justify=13):
             gandi.separator_line()
             output_generic(gandi, schedule, schedule_keys, justify)
 
-
-def check_domain_available(ctx, domain):
+def check_domain_available(ctx, pointless_click_placeholder, domain):
     """ Helper to check if a domain is available."""
+    # pointless_click_placeholder is to prevent Click compatibility warning
     gandi = ctx.obj
+    if domain is None:
+        domain = click.prompt("Enter the domain name you want to register")
```

Comment:
you should not make a prompt in a low level func like a callback here.

Anyway if domain  is now an argument then it's mandatory so you will always have a domain when getting inside this callback, so this test is not useful and you can remove it